app/models/mixins/per_ems_worker_mixin.rb

```diff
@@ -101,10 +101,19 @@ def queue_name_for_ems(ems)
     end
 
     def ems_id_from_queue_name(queue_name)
-      return nil if queue_name.blank?
-      name, id = queue_name.split("_")
-      return nil unless name == "ems"
-      id.to_i
+      if queue_name.kind_of?(Array)
+        queue_names = []
```

Comment:
This should be `ems_ids` instead of `queue_names` since this block is returning a list of `ems_ids`